yb-voyager/cmd/live_migration_integration_test.go

```diff
@@ -0,0 +1,711 @@
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	// Connect to both Postgres and YugabyteDB.
+	pgConn, err := postgresContainer.GetConnection()
+	testutils.FatalIfError(t, err, "connecting to Postgres")
+
+	ybConn, err := yugabytedbContainer.GetConnection()
+	testutils.FatalIfError(t, err, "Error connecting to YugabyteDB")
+
+	// Compare the full table data between Postgres and YugabyteDB for snapshot part.
+	// We assume the table "test_data" has a primary key "id" so we order by it.
+	if err := testutils.CompareTableData(ctx, pgConn, ybConn, "test_schema.test_live", "id"); err != nil {
+		t.Errorf("Table data mismatch between Postgres and YugabyteDB: %v", err)
+	}
+
+	//streaming events 15 events
+	postgresContainer.ExecuteSqls([]string{
+		`INSERT INTO test_schema.test_live (name, email, description)
+SELECT
+	md5(random()::text),                                      -- name
+	md5(random()::text) || '@example.com',                    -- email
+	repeat(md5(random()::text), 10)                           -- description (~320 chars)
+FROM generate_series(1, 15);`,
+	}...)
+
+	start = time.Now()
+	for time.Since(start) < (30 * time.Second) { // 30 seconds timeout
+		if streamingPhaseCompleted(t, postgresContainer.GetConfig().Password,
+			yugabytedbContainer.GetConfig().Password, 15, `test_schema."test_live"`) {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	// Compare the full table data between Postgres and YugabyteDB for streaming part.
+	if err := testutils.CompareTableData(ctx, pgConn, ybConn, "test_schema.test_live", "id"); err != nil {
+		t.Errorf("Table data mismatch between Postgres and YugabyteDB after streaming: %v", err)
+	}
+
+	//Stopping import command
+	if err := importCmd.Kill(); err != nil {
+		testutils.FatalIfError(t, err, "killing the import data process errored")
+	}
+
+	time.Sleep(5 * time.Second)
+
+	// Wait for the command to exit.
+	if err := importCmd.Wait(); err != nil {
+		t.Logf("Async import run exited with error (expected): %v", err)
+	} else {
+		t.Logf("Async import run completed unexpectedly")
+	}
+
+	time.Sleep(10 * time.Second)
+
+	// Perform cutover
+	err = testutils.NewVoyagerCommandRunner(nil, "initiate cutover to target", []string{
+		"--export-dir", exportDir,
+		"--yes",
+		"--prepare-for-fall-back", "false",
+	}, nil, false).Run()
+	testutils.FatalIfError(t, err, "Cutover command failed")
+
+	//Resume import command to finish the cutover
+	err = testutils.NewVoyagerCommandRunner(yugabytedbContainer, "import data", []string{
+		"--export-dir", exportDir,
+		"--disable-pb", "true",
+		"--yes",
+	}, nil, true).Run()
+	testutils.FatalIfError(t, err, "import data failed")
+
+	metaDB, err = metadb.NewMetaDB(exportDir)
+	testutils.FatalIfError(t, err, "Failed to initialize meta db")
+
+	start = time.Now()
+	for time.Since(start) < (30 * time.Second) { // 30 seconds timeout
+		status := getCutoverStatus()
+		if status == COMPLETED {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
```

Comment:
The loop is being used at multiple place
I still think a helper util function for this will be better
where you provide how much timeout, and a closure function.